silkrpc/common/log.hpp

```diff
@@ -20,22 +20,43 @@
 #include <absl/strings/string_view.h>
 
 #include <string>
+#include <mutex>
 
-#include <silkworm/common/log.hpp>
+namespace silkrpc {
 
-namespace silkworm {
+// available verbosity levels
+enum class LogLevel { Trace, Debug, Info, Warn, Error, Critical, None };
 
-bool AbslParseFlag(absl::string_view text, LogLevel* level, std::string* error);
-std::string AbslUnparseFlag(LogLevel level);
+// silence
+std::ostream& null_stream();
 
-} // namespace silkworm
+//
+// Below are for access via macros ONLY.
+// Placing them in detail namespace prevents use of macros in nested namespace
+//
+extern LogLevel log_verbosity_;
+extern bool log_thread_enabled_;
+void log_set_streams_(std::ostream& o1, std::ostream& o2);
+class log_ {
+  public:
+    explicit log_(LogLevel level) : level_(level) { log_mtx_.lock(); }
+    ~log_() { log_mtx_.unlock(); }
+    std::ostream& header_(LogLevel);
+    template <class T>
+    std::ostream& operator<<(const T& message) {
+        return header_(level_) << message;
+    }
+
+  private:
+    LogLevel level_;
+    static std::mutex log_mtx_;
+};
 
-namespace silkrpc {
 
-using Logger = silkworm::log_;
-using LogLevel = silkworm::LogLevel;
+using Logger = log_;
+using LogLevel = LogLevel;
 
-#define LOG(level_) if ((level_) < silkworm::log_verbosity_) {} else silkworm::log_(level_) << " " // NOLINT
+#define LOG(level_) if ((level_) < silkrpc::log_verbosity_) {} else silkrpc::log_(level_) << " " // NOLINT
```

Comment:
We're already in `silkrpc` namespace:

`silkrpc::log_verbosity_` -> `log_verbosity_`
`silkrpc::log_` -> `log_`

Reply:
Wrong